src/nerve_tools/configure_nerve.py

```diff
@@ -142,14 +145,20 @@ def generate_subconfiguration(
                         'headers': healthcheck_headers,
                     },
                 ],
-                'labels': labels,
+                'labels': default_labels,
             }
 
-            v2_key = '%s.%s:%d.v2.new' % (
-                service_name, zk_location, port,
+            v2_key = '%s.%s:%s.%d.v2.new' % (
+                service_name, zk_location, loc, port,
             )
 
-            if v2_key not in config:
+            if v2_key not in config and typ == most_specific_advertise:
```

Comment:
Let's leave a comment here explaining that we're explicitly relying on Synapse to deduplicate identical backends.

https://github.com/airbnb/synapse/blob/master/lib/synapse/service_watcher/base.rb#L136